packages/nimble-components/src/wafer-map/specs/features/change-highlight.md

```diff
@@ -0,0 +1,230 @@
+
+The component can highlight only based on the value that appears on the die. For example here:
+
+![dies-values](resources/dies-values.PNG)
+
+We wish to not be constrained by the values appearing on the die but to be able to highlight based on what we receive from the selection function
+
+## Links To Relevant Work Items and Reference Material
+
+[MicroStrategy Enable Custom Selection and Highlighting](https://www2.microstrategy.com/producthelp/Current/VisSDK/Content/topics/HTML5/selection_highlightAPI.htm)
+
+[Introduction to Attributes, Metrics, and Groups](https://www2.microstrategy.com/producthelp/Current/MSTRWeb/WebHelp/Lang_1033/Content/CreatingNewObjects.htm)
+
+## Implementation / Design
+
+We will replace the old logic, removing it completely, and add a new one.
+
+Each die will have an optional field called "tags", being optional in the cases when we don't use highlighting we can get rid of it.
+
+```ts
+export interface WaferMapDie{
+...
+   tags?: string[];
+...
+}
+```
+
+How the tags looks like remain a choice of the user, they can either have a simplified form
+
+```ts
+['a', 'b', 'c'];
+```
+
+Or can be the id of the attributes
+
+```ts
+[
+    'h90;85C4BC04377947939B9B8E62A7AC9BA5',
+    'h530;208BD62A39A84D96825CEBF10B1E6179',
+    'h80;A1DEB28673CA4A2B8C8DE26461D2FFB7'
+];
+```
+
+We will make the highlights based on this tags. The wafer will receive a list of highlightedTags
+
+```ts
+@observable public highlightedTags: string[]
+```
+
+Example of flow :
+
+```ts
+export const dies: WaferMapDie[] = [
+    {
+        x: 0,
+        y: 0,
+        value: '14',
+        tags: ['a']
+    },
+    {
+        x: 1,
+        y: 0,
+        value: '76',
+        tags: ['a', 'b']
+    },
+    {
+        x: 0,
+        y: 1,
+        value: '72'
+    }
+];
+```
+
+A die will be highlighted if any of the die's tags equals at least one value in the highlightedTags
+
+If the highlightedTags equals
+
+```ts
+['a'];
+```
+
+dies[0][0] will be highlighted because it contains 'a'
+
+dies[1][0] will be highlighted because it also contains 'a'
+
+dies[0][1] will not be be highlighted because it does not have any tag
+
+```ts
+['b', 'h'];
+```
+
+dies[0][0] will not be highlighted because it does not contain 'b' neither 'h'
+
+dies[1][0] will be highlighted because it contains 'b'
+
+dies[0][1] will not be be highlighted because it does not have any tag
+
+If the highlightedTags equals
+
+```ts
+[];
+```
+
+Nothing will be highlighted
+
+If the highlightedTags equals
+
+```ts
+['efg'];
+```
+
+All dies will be faded
+
+Note: Besides the code changes we will also need to update the tests and nimble story book.
+
+## Alternative Designs
+
+1. Same as the main one but instead of using a strings array for highlighted tags use an matrix of strings. This will give us more flexibility and more highlight capabilities but it will also require more time and will increase the complexity of complexity of the highlighting. In this case:
```

Comment:
The numbered list does not render in markdown well. You would need to indent all the paragraphs for them to show up in the list. CCan youu instead make these different sub headings?

```
## Alternative Designs

### Highlighted tags with a matrix of strings

Same as the main one but instead of using a strings...

### Add an isHighlighted field to dies

Introduce a new logic of highlighting...
```